build.gradle

```
@@ -107,23 +134,56 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
-class ResolveIntelliJVersions extends DefaultTask {
-  @Input
-  @Option(option = 'version-key', description = 'Either release number (like 2020.3) or key of intellijVersions (like eapOfLatestSupportedMajor)')
-  String versionKey
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Document
+
+String findNewestEapBuildNumber() {
+  Document htmlContent = Jsoup.connect(project.intellijSnapshotsUrl).get()
+  Element link = htmlContent.select('a[href^='+project.intellijSnapshotsUrl+'com/jetbrains/intellij/idea/ideaIC/][href$="-EAP-SNAPSHOT.pom"]')[0]
+  if (link) {
+    String attr = link.attr("href")
+    Pattern pattern = Pattern.compile("ideaIC-(.*?)-EAP-SNAPSHOT.pom")
+    Matcher matcher = pattern.matcher(attr)
+    matcher.find()
+    return matcher.group(1)
+  }
+  return null
+}
 
-  @OptionValues('version-key')
-  List<String> getAvailableVersionKeys() {
-    return new ArrayList<>(project.intellijVersions.keySet())
+String checkForEapOfFutureRelease(String intellijMajorPartOfBuildNumber) {
+  String foundBuildNumber = project.findNewestEapBuildNumber()
+  if (foundBuildNumber) {
+    if (Integer.parseInt(majorPartOfBuildNumber(foundBuildNumber)) > Integer.parseInt(intellijMajorPartOfBuildNumber)) {
+      return foundBuildNumber
+    }
   }
+  return null
+}
 
-  @TaskAction
-  void print() {
-    println(String.join("\n", project.resolveIntelliJVersions(versionKey)))
+String checkForEapWithBuildNumberHigherThan(String latestEapBuildNumber) {
+  String foundBuildNumber = project.findNewestEapBuildNumber()
+  if (foundBuildNumber) {
+    if (VersionNumber.parse(foundBuildNumber) > VersionNumber.parse(latestEapBuildNumber)) {
+      return foundBuildNumber
+    }
   }
+  return null
 }
 
-tasks.register('resolveIntelliJVersions', ResolveIntelliJVersions)
+task updateEapBuildNumber {
+  Properties properties = project.intellijVersionsProperties()
+  File propFile = project.intellijVersionsFile()
+  String latestEapBuildNumber = project.intellijVersions.eapOfLatestSupportedMajor
+  String newerEapBuildNumber = latestEapBuildNumber ? project.checkForEapWithBuildNumberHigherThan(latestEapBuildNumber.split('-')[0]) :
+            project.checkForEapOfFutureRelease(intellijVersionToBuildNumber(project.intellijVersions.latestStable))
```

Would something like that work? 🤔 
```suggestion
  String newerEapBuildNumber = latestEapBuildNumber ? project.checkForEapWithBuildNumberHigherThan(latestEapBuildNumber.split('-')[0]) :
            project.checkForEapWithBuildNumberHigherThan(intellijVersionToBuildNumber(project.intellijVersions.latestStable) + '.9999999.9999999')
```